src/server/render.js

```diff
@@ -48,6 +49,27 @@ const normalizeRender = vm => {
   }
 }
 
+function waitForSsrPrefetch (vm, resolve, reject) {
+  let handlers = vm.$options.ssrPrefetch
+  if (isDef(handlers)) {
+    if (!Array.isArray(handlers)) handlers = [handlers]
+    try {
+      const promises = []
+      for (let i = 0, j = handlers.length; i < j; i++) {
```

Comment:
@Akryum pardon me the intrusion -- just thought i'd [drop this here][https://medium.com/kbdev/voyage-to-the-most-efficient-loop-in-nodejs-and-a-bit-js-5961d4524c2e] -- I see that you're using a pattern for `for` loops that I used to employ myself, but, unless you assign the length to an external variable, this idiom actually makes things slower :(